sources/api/openapi.yaml

```diff
@@ -324,3 +324,79 @@ paths:
           description: "Reboot requested"
         500:
           description: "Server error"
+
+  /actions/refresh-update:
+    post:
+      summary: "Query update repository and refresh list of updates"
+      operationId: "refresh_update"
+      responses:
+        204:
+          description: "Successful request"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/prepare-update:
+    post:
+      summary: "Downloads the chosen update and write the update image to the inactive partition"
+      operationId: "prepare_update"
+      responses:
+        204:
+          description: "Successful request"
+        404:
+          description: "Chosen update does not exist"
+        409:
+          description: "Action not allowed according to current update state"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/activate-update:
+    post:
+      summary: "Marks the inactive partition for boot to apply update"
+      operationId: "activate_update"
+      responses:
+        204:
+          description: "Successfully activated update"
+        404:
+          description: "No update image applied to inactive partition"
```

Comment:
I'm having a hard time following the error conditions between apiserver and t-b-u... when exactly would activate-update give a 404, can we clarify?

Does it imply the inactive partition is empty?  Or do we not allow people to call this API unless we applied an update through t-b-u, and forbid using it to switch to the other partition if it was applied directly through updog or otherwise?

(I think it's the latter because of the state machine, hence why deactivate-update exists... if activate-update always flipped banks, we wouldn't need a separate "deactivate".)

Reply:
Yeah it's the latter.
If `StagingPartition` in the update status is `null`, then from `t-b-u`'s perspective, there is no update to activate so it would return a 404. (This should theoretically never happen because of the state enforcement on commands)